lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+
+  defstruct(
+    players: [],
+    npcs: [],
+    map: nil)
+
+  def join(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_join, player_entity})
+  end
+
+  def leave(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_leave, player_entity})
+  end
+
+  def register(entity, map, players=[], npc_info=[]),
+  do: Entity.put_behaviour(entity, MapInstance.Behaviour, map, players, npc_info)
+
+  def register(entity, map_instance),
+  do: Entity.put_behaviour(entity, MapInstance.Behaviour, map_instance)
+
+  def unregister(entity),
+  do: Entity.remove_behaviour(entity, MapInstance.Behaviour)
+
+  defmodule Behaviour do
+    use Entice.Entity.Behaviour
+
+    def init(entity, map, players, npc_info) do
```

Comment:
wont match. init expects 2 params, so you will need to box thos additional fields in a tuple or so